### PR TITLE
Benchmarks: Convert Benchmarks.EF6 to use BenchmarksDotNet

### PR DESCRIPTION
**Pull request by smitpatel**
Move benchmarks to separate directory
Create Separate solution for benchmarks of older releases
Don't build Benchmarks.sln while build.cmd script

First part of #8842

### CODE REVIEW

**Comment by smitpatel** [comment]
Looks like build fails.

**Comment by smitpatel** [comment]
Build passes.

File: benchmarks/EFCore.Benchmarks.EF6/ChangeTracker/DbSetOperationTests.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                Context.Dispose();
+            }
+        }
+
+        public class AddDataVariations : Base
+        {
+            [IterationSetup]
+            public void InitializeContext()
+            {
+                Context = Fixture.CreateContext();
```
Comment (anpete) [inline benchmarks/EFCore.Benchmarks.EF6/ChangeTracker/DbSetOperationTests.cs]: Do this in Base?

Comment (smitpatel) [inline benchmarks/EFCore.Benchmarks.EF6/ChangeTracker/DbSetOperationTests.cs]: Marking all methods virtual. & doing stuff in base wherever common.

File: benchmarks/EFCore.Benchmarks.EF6/Program.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            benchmarkSummaryProcessor.Process(BenchmarkRunner.Run<NavigationsQueryTests>());
+            benchmarkSummaryProcessor.Process(BenchmarkRunner.Run<QueryCompilationTests>());
+            benchmarkSummaryProcessor.Process(BenchmarkRunner.Run<RawSqlQueryTests>());
+            benchmarkSummaryProcessor.Process(BenchmarkRunner.Run<SimpleQueryTests>());
+
+            // Update
+            benchmarkSummaryProcessor.Process(BenchmarkRunner.Run<SimpleUpdatePipelineTests.Insert>());
+            benchmarkSummaryProcessor.Process(BenchmarkRunner.Run<SimpleUpdatePipelineTests.Update>());
+            benchmarkSummaryProcessor.Process(BenchmarkRunner.Run<SimpleUpdatePipelineTests.Delete>());
+            benchmarkSummaryProcessor.Process(BenchmarkRunner.Run<SimpleUpdatePipelineTests.Mixed>());
```
Comment (bricelam) [inline benchmarks/EFCore.Benchmarks.EF6/Program.cs]: Would be nice to not have to update this every time we add a new test.

Comment (smitpatel) [inline benchmarks/EFCore.Benchmarks.EF6/Program.cs]: Discussed with @bricelam . This allows us to run particular test locally if we want to run. Whenever adding new benchmarks you need to add line here to test anyway so keeping this as is also have value.

**Comment by smitpatel** [comment]
Reworking a bit (converting EFCore dev tests first to compare with current numbers first)

**Pull request closed without merging**